video_xblock/static/js/studio-edit-transcripts-manual-upload.js

```diff
@@ -83,6 +100,57 @@ function validateTranscripts(e, $langChoiceItem) {
 }
 
 /**
+ * Validate extension and name of a transcript or a caption file before save it to video xblock.
+ *
+ * Returns:
+ * isValidated (Boolean): Result of a validation (true|false).
+ *
+ */
+function validateTranscriptFile(event, fieldName, filename, $fileUploader) {
+    'use strict';
+    // User may upload a file without extension. Reference: http://stackoverflow.com/a/1203361
+    var fileExtension = filename.substr((~-filename.lastIndexOf('.') >>> 0) + 2);
```

Comment:
In this case, we can sacrifice extreme corner cases handling and some performance in the name of readability.

`filename.split('.').pop()` is better fit here.